packages/editor/src/components/block-list/block.js

```diff
@@ -403,7 +416,8 @@ export class BlockListBlock extends Component {
 					// We render block movers and block settings to keep them tabbale even if hidden
 					const shouldRenderMovers = ! isFocusMode && ( isSelected || hoverArea === 'left' ) && ! showEmptyBlockSideInserter && ! isMultiSelecting && ! isPartOfMultiSelection && ! isTypingWithinBlock;
 					const shouldShowBreadcrumb = ! isFocusMode && isHovered && ! isEmptyDefaultBlock;
-					const shouldShowContextualToolbar = ! hasFixedToolbar && ! showSideInserter && ( ( isSelected && ( ! isTypingWithinBlock || isCaretWithinFormattedText ) ) || isFirstMultiSelected );
+					const contextualToolbarMayAppear = ! hasFixedToolbar && ! showSideInserter;
```

Comment:
I'm wondering if this really ought to be a condition including `showSideInserter`, or if it's enough to render `KeyboardShortcuts` with direct consideration of `! hasFixedToolbar`. Thoughts?

Reply:
Hi @aduth,
I added a new condition, that specifies in which cases we should render KeyboardShortcuts more explicitly.
We should take into account the isEmptyDefaultBlock flag, if true we should not use the shortcut as in this case the block has no toolbar, and we should continue to focus the header toolbar.
Let me know you prefer this condition; it should be equivalent to the previous one, just in a different format.

Reply:
I like it 👍